### DISCUSSION TITLE
Allegedly wrong renovate configuration

### DISCUSSION THREAD

**Discussion by bitjoo** [renovatebot/renovate#13641] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

31.28.3

### Please select which platform you are using if self-hosting.

GitHub Enterprise Server

### Describe the bug

In my repository renovate creates an issue called **Action Required: Fix Renovate Configuration**

It says: 

> There is an error with this repository's Renovate configuration that needs to be fixed. As a precaution, Renovate will stop PRs until it is resolved.

> File: `renovate.json`
Error type: The renovate configuration file contains some invalid settings
Message: `Invalid configuration option: packageRules[0].matchPackagePatterns, Invalid configuration option: packageRules[1].matchPackagePatterns, Invalid configuration option: packageRules[2].matchPackagePatterns, Invalid configuration option: packageRules[3].matchPackagePatterns, packageRules: Each packageRule must contain at least one selector (paths, languages, baseBranchList, managers, datasources, depTypeList, packageNames, packagePatterns, excludePackageNames, excludePackagePatterns, sourceUrlPrefixes, updateTypes, matchCurrentVersion). If you wish for configuration to apply to all packages, it is not necessary to place it inside a packageRule at all., packageRules: Each packageRule must contain at least one selector (paths, languages, baseBranchList, managers, datasources, depTypeList, packageNames, packagePatterns, excludePackageNames, excludePackagePatterns, sourceUrlPrefixes, updateTypes, matchCurrentVersion). If you wish for configuration to apply to all packages, it is not necessary to place it inside a packageRule at all.`

I am using `renovate 31.28.3` and instead of the deprecated `packagePatterns` I use `matchPackagePatterns` in my config.
So either my configuration uses deprecated options then renovate warns me during the run, but this issue isn't created or I am using the latest recommended options and get this issue. 

### Relevant debug logs

<details><summary>Logs</summary>

```
File: `renovate.json`
Error type: The renovate configuration file contains some invalid settings
Message: `Invalid configuration option: packageRules[0].matchPackagePatterns, Invalid configuration option: packageRules[1].matchPackagePatterns, Invalid configuration option: packageRules[2].matchPackagePatterns, Invalid configuration option: packageRules[3].matchPackagePatterns, packageRules: Each packageRule must contain at least one selector (paths, languages, baseBranchList, managers, datasources, depTypeList, packageNames, packagePatterns, excludePackageNames, excludePackagePatterns, sourceUrlPrefixes, updateTypes, matchCurrentVersion). If you wish for configuration to apply to all packages, it is not necessary to place it inside a packageRule at all., packageRules: Each packageRule must contain at least one selector (paths, languages, baseBranchList, managers, datasources, depTypeList, packageNames, packagePatterns, excludePackageNames, excludePackagePatterns, sourceUrlPrefixes, updateTypes, matchCurrentVersion). If you wish for configuration to apply to all packages, it is not necessary to place it inside a packageRule at all.`
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
Check the logs, I'm pretty sure you are not using latest version of renovate. Renovate writes the used version to the logs.

How are you run renovate?

**Reply by bitjoo** [reply]
I checked the logs and they told me:
```
11:00:22  renovate_1  |  INFO: Repository started (repository=xxxxxxxxxxxx)
11:00:22  renovate_1  |        "renovateVersion": "31.28.3"
```

I am using the docker image from https://hub.docker.com/r/renovate/renovate and run it via jenkins.
I mount the `config.json` in the docker container.

**Reply by rarkins** [reply]
Please run with `LOG_LEVEL=debug` in your env and copy the full log here when you get this error

**Reply by bitjoo** [reply]
Here is the full log. I have made some parts unrecognizable. So don't be confused about the `xxxxx` you will find there.

[jenkins.log](https://github.com/renovatebot/renovate/files/7903663/jenkins.log)

**Reply by rarkins** [reply]
I cannot find the string "The renovate configuration file contains some invalid settings" in this log

**Reply by bitjoo** [reply]
This is never shown in the log. It's only visible in the issue that is opened and closed constantly.

**Reply by rarkins** [reply]
The log shows multiple types of problems. Which line would you like to focus on?

**Reply by bitjoo** [reply]
I am not sure, what exactly the problems are. Perhaps you can point me to one and we can have a look, what might be wrong.

But I am still confused about the initial mentioned error message. Do you think there is some kind of renovate version mismatch? How is that possible since it runs only in a docker container with the latest version?

**Reply by rarkins** [reply]
Sorry, but I don't have time for that
